src/main/java/uk/gov/hmcts/ccd/data/casedataaccesscontrol/DefaultRoleAssignmentRepository.java

```diff
@@ -208,6 +218,37 @@ public RoleAssignmentResponse findRoleAssignmentsByCasesAndUsers(List<String> ca
         }
     }
 
+    private RoleAssignmentResponse findRoleAssignmentsByCasesAndUsers(RoleAssignmentQuery roleAssignmentQuery) {
+        RoleAssignmentResponse roleAssignmentResponse = new RoleAssignmentResponse();
+        roleAssignmentResponse.setRoleAssignments(new ArrayList<>());
+        final HttpHeaders headers = securityUtils.authorizationHeaders();
+
+        int pageNumber = ROLE_ASSIGNMENT_STARTING_PAGE_NUMBER;
+        int pageSize = Integer.parseInt(applicationParams.getRoleAssignmentPageSize());
+        int totalRecords;
+
+        do {
+            headers.add(ROLE_ASSIGNMENT_PAGE_NUMBER_HEADER, Integer.toString(pageNumber));
+            headers.add(ROLE_ASSIGNMENT_PAGE_SIZE_HEADER, Integer.toString(pageSize));
+            final var requestEntity = new HttpEntity(roleAssignmentQuery, headers);
+
+            ResponseEntity<RoleAssignmentResponse> responseEntity = restTemplate.exchange(
+                applicationParams.amQueryRoleAssignmentsURL(),
+                HttpMethod.POST,
+                requestEntity,
+                RoleAssignmentResponse.class);
+
+            totalRecords = Integer.parseInt(Objects.requireNonNull(Objects.requireNonNull(
```

Comment:
Why nesting like Objects.requireNonNull(Objects.requireNonNull ?

Reply:
We need this to handle null checks for two different fields

Reply:
done